tests/apollo/test_skvbc_block_accumulation.py

```diff
@@ -0,0 +1,149 @@
+BATCH_BY_REQ_NUM = "2"
+MAX_REQ_NUM_IN_BATCH = 5
+BATCH_FLUSH_PERIOD = "250"
+
+def start_replica_cmd(builddir, replica_id, view_change_timeout_milli="10000"):
+    """
+    Return a command that starts an skvbc replica when passed to
+    subprocess.Popen.
+
+    The replica is started with a short view change timeout.
+
+    Note each arguments is an element in a list.
+    """
+
+    status_timer_milli = "500"
+
+    path = os.path.join(builddir, "tests", "simpleKVBC", "TesterReplica", "skvbc_replica")
+    return [path,
+            "-k", KEY_FILE_PREFIX,
+            "-i", str(replica_id),
+            "-s", status_timer_milli,
+            "-v", view_change_timeout_milli,
+            "-b", BATCH_BY_REQ_NUM,
+            "-q", str(MAX_REQ_NUM_IN_BATCH),
+            "-z", BATCH_FLUSH_PERIOD,
+            "-o", str(True)
+            ]
+
+def start_replica_cmd_with_vc_timeout(vc_timeout):
+    def wrapper(*args, **kwargs):
+        return start_replica_cmd(*args, **kwargs, view_change_timeout_milli=vc_timeout)
+    return wrapper
+
+
+class SkvbcBlockAccumulationTest(unittest.TestCase):
+
+    __test__ = False  # so that PyTest ignores this test scenario
+
+    async def send_single_batch_write_with_pre_execution_and_kv(self, skvbc, client, batch_size, long_exec=False, ):
+        msg_batch = []
+        batch_seq_nums = []
+        dic_writeset = {}
+        final_block_ids = set()
+        for i in range(batch_size):
+            readset = set()
+            writeset = self.writeset(skvbc, 2, dic_writeset)
+            msg_batch.append(skvbc.write_req(readset, writeset, 0, long_exec))
+            seq_num = client.req_seq_num.next()
+            batch_seq_nums.append(seq_num)
+        replies = await client.write_batch(msg_batch, batch_seq_nums)
+        for seq_num, reply_msg in replies.items():
+            self.assertTrue(skvbc.parse_reply(reply_msg.get_common_data()).success)
+            reply = skvbc.parse_reply(reply_msg.get_common_data())
+            final_block_ids.add(reply.last_block_id)
+        return final_block_ids,dic_writeset
+
+    def writeset(self, skvbc, max_size, dic_writeset, keys=None):
+       write_set = [(skvbc.unique_random_key() , skvbc.random_value()),
+                    (skvbc.unique_random_key() , skvbc.random_value())]
+       dic_writeset.update(write_set)
+       return write_set
+
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    @with_constant_load
+    async def test_batch_block_accumulation_request_block_write_validation(self, bft_network, skvbc, constant_load):
+        """
+        Launch a batch request and validate the block write keys.
+        """
+        bft_network.start_all_replicas()
+        await trio.sleep(SKVBC_INIT_GRACE_TIME)
+        await bft_network.init_preexec_count()
+
+        client = bft_network.random_client()
+        client.config = client.config._replace(
+            req_timeout_milli=LONG_REQ_TIMEOUT_MILLI,
```

Comment:
Why LONG_REQ_TIMEOUT_MILLI is used here? It should SHORT_REQ_TIMEOUT_MILLI.

Reply:
updated test script with modified configuration